pkg/ethereum/node.go

```diff
@@ -252,16 +292,14 @@ func (n *Node) addToBackoffCache(pid peer.ID, addrInfo peer.AddrInfo) {
 }
 
 func (n *Node) removeFromBackoffCache(pid peer.ID) {
-	n.cacheMutex.Lock()
-	defer n.cacheMutex.Unlock()
-
+	// NOTE: Removing the peer so no need to lock the cache
```

Comment:
The mutex here is concurrent read-write protection for the whole map. If we're reading / writing to this map concurrently, then all operations require a mutex lock, including deletes

Reply:
One thing to note - I notice that the handshake logs get blocked/clogged when I add the mutex lock, `removeFromBackoffCache` is called inside `addToMetadataCache` which is called after publishing metadata. I am suspecting the reason could be this

Reply:
Added this for now, can investigate later if the issue persists

Reply:
Yeah let's keep this open until we know